pkg/descheduler/strategies/lownodeutilization.go

```diff
@@ -274,28 +258,24 @@ func evictPodsFromTargetNodes(
 	sortNodesByUsage(targetNodes)
 
 	// upper bound on total number of pods/cpu/memory to be moved
-	totalAvailableUsage := map[v1.ResourceName]*resource.Quantity{
-		v1.ResourcePods:   {},
-		v1.ResourceCPU:    {},
-		v1.ResourceMemory: {},
-	}
+	totalAvailableUsage := map[v1.ResourceName]*resource.Quantity{}
 
 	var taintsOfLowNodes = make(map[string][]v1.Taint, len(lowNodes))
 	for _, node := range lowNodes {
 		taintsOfLowNodes[node.node.Name] = node.node.Spec.Taints
 
-		for name := range totalAvailableUsage {
+		for name := range node.highResourceThreshold {
+			if _, ok := totalAvailableUsage[name]; !ok {
+				totalAvailableUsage[name] = resource.NewMilliQuantity(0, resource.DecimalSI)
```

Comment:
You can keep `totalAvailableUsage` as it as since pods, cpu and memory are supposed to be always present. Just extended `totalAvailableUsage` with new resources. Also, `resource.NewQuantity` is more preferable as extended resources are natural numbers.